service/history/service.go

```diff
@@ -432,10 +434,50 @@ func (s *Service) Stop() {
 		return
 	}
 
+	// initiate graceful shutdown :
+	// 1. remove self from the membership ring
+	// 2. wait for other members to discover we are going down
+	// 3. stop acquiring new shards (periodically or based on other membership changes)
+	// 4. wait for shard ownership to transfer (and inflight requests to drain) while still accepting new requests
+	// 5. reject all requests arriving at rpc handler to avoid taking on more work
```

Comment:
nit: probably worth documenting which requests will still be accepted at this stage.

Reply:
👍